src/server/utils/logging/otel/formatter.js

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright 2023 American Express Travel Related Services Company, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+import os from 'os';
+import util from 'util';
+import { SeverityNumber } from '@opentelemetry/api-logs';
+import flatten from 'flat';
+
+function getBaseRecord(level) {
+  let severityText = level.toUpperCase();
+  if (!Object.keys(SeverityNumber).includes(severityText)) severityText = 'INFO';
```

Comment:
could we use a `Map` for a cache? (avoid the `.keys` and `.includes` loops)
maybe
```suggestion
const levelToSeverityText = new Map();

function getBaseRecord(level) {
  let severityText = levelToSeverityText.get(level);
  if (!severityText) {
    severityText = level.toUpperCase();
    if (!Object.keys(SeverityNumber).includes(severityText)) {
      severityText = 'INFO';
    }
    levelToSeverityText.set(level, severityText);
  }
```

lots of profiling/measuring opportunities for the hot paths 😅 (esp. to check confidence that these are even notable performance impacts)

Reply:
```suggestion
or maybe
function getBaseRecord(level) {
  let severityText = level.toUpperCase();
  if (typeof SeverityNumber[severityText] === 'undefined') {
    severityText = 'INFO';
  }
```